cli/src/commands/obslog.rs

```diff
@@ -113,7 +113,11 @@ pub(crate) fn cmd_obslog(
     let mut commits = topo_order_reverse_ok(
         vec![Ok(start_commit)],
         |commit: &Commit| commit.id().clone(),
-        |commit: &Commit| commit.predecessors().collect_vec(),
+        |commit: &Commit| {
+            let mut predecessors = commit.predecessors().collect_vec();
```

Comment:
nit: Can you add inline comment why walking from the second (or the last)? If I understand the idea, we assume that the first predecessor usually has longer history.

Reply:
I struggled a bit with explaining my reasoning, so I added a graph example to try to clarify it. I'm not sure if it's too long of a comment now or if there's some easier way to explain it. Could you take a look and let me know what you think?

My main idea is that we want to show more recent changes towards the top of the log and less recent changes towards the bottom of the log, and the old iteration order was backwards for squashed commits (the most recent squashes appeared at the bottom of the log).

Reply:
Okay, if the intent is to emit recent rewrites first, something like "visit from last because predecessors are usually recorded in chronological order on squash" will help.

My concern here was that `predecessors` has no defined order, so it's unclear what's our assumption.

Reply:
Ok I rewrote it to include that explanation so hopefully it's a bit clearer now. Let me know what you think!